test/tests/functional/pbs_reliable_job_startup.py

```diff
@@ -0,0 +1,4473 @@
+                chunk (except the MS (first) chunk), resulting in the
+                assignment:
+
+                    exec_vnode = (A)+(B)+(D)+(C)+(E)
+
+                where D mirrors super-chunk B specs while E mirrors
+                chunk C.
+
+             3. Prior to submitting a job, suspend mom B. When job runs,
+                momB won't be able to join the job, so it won't be considered
+                as a "healthy" mom.
+
+             4. Have an execjob_prologue hook that doesn't fail any mom host
+                with alarm=alarm1, order=1.
+
+             5. Have an execjob_prologue hook2 with alarm=alarm2, order=2,
+                that fails (causes rejection) when executed by mom managing
+                vnodes in (C).
+
+             6. Before job runs, kill -STOP mom (B).
+
+             7. Have an execjob_launch hook that prunes back the
+                job's exec_vnode assignment back to satisfying the original
+                3-node select spec, choosing only healthy nodes.
+
+             8. Result:
+
+                a. This results in the following reassignment of chunks:
+
+                   exec_vnode = (A)+(D)+(E)
+
+                   since (B) and (C) contain vnodes from failed moms.
+
+                b. vnodes in (B) and (C) are now showing a state of "free".
+
+                c. Mom's log file shows the wait time from execjob_prologue
+                   hook1 execution and the execution of the exescjob_launch
+                   hook is no more than alarm1+alarm2.
+
+                c. The accounting log start record 'S' will reflect the
+                   select request where additional chunks were added, while
+                   the secondary start record 's' will reflect the assigned
+                   resources after pruning the original select request via
+                   the pbs.release_nodes(keep_select=...) call
+                   inside execjob_launch hook.
+        """
+
+        # instantiate queuejob hook
+        hook_body = """
+import pbs
+e=pbs.event()
+pbs.logmsg(pbs.LOG_DEBUG, "queuejob hook executed")
+# Save current select spec in resource 'site'
+e.job.Resource_List["site"] = str(e.job.Resource_List["select"])
+# Increment chunks by 1
+new_select = e.job.Resource_List["select"].increment_chunks(1)
+e.job.Resource_List["select"] = new_select
+e.job.tolerate_node_failures = "job_start"
+"""
+        hook_event = "queuejob"
+        hook_name = "qjob"
+        a = {'event': hook_event, 'enabled': 'true'}
+        self.server.create_import_hook(hook_name, a, hook_body)
+
+        # instantiate execjob_begin hook
+        hook_body = """
+import pbs
+e=pbs.event()
+pbs.logmsg(pbs.LOG_DEBUG, "Executing begin")
+localnode=pbs.get_local_nodename()
+"""
+        hook_event = "execjob_begin"
+        hook_name = "begin"
+        a = {'event': hook_event, 'enabled': 'true'}
+        self.server.create_import_hook(hook_name, a, hook_body)
+
+        # instantiat
```

Comment:
In CLI mode, maybe.  In API mode, this will fail.  This is why there is a PTL interface for attributes like this.  Look at pbs_ifl_mock.py and pbs_api_to_cli.py

Reply:
Ok, I've updated pbs_ifl_mock.py and pbs_api_to_cli.py so that the following now works:
a = {ATTR_tolerate_node_failures: "none"}
self.server.alterjob(jobid=jid, attrib=a)